### DISCUSSION TITLE
In MONAI's focal loss implementation, why does the alpha parameter become invalid when softmax is used and include_background=False?

### DISCUSSION THREAD

**Discussion by dancingreindeer** [Project-MONAI/MONAI#8116] [Q&A]
```python
if self.use_softmax: 
   if not self.include_background and self.alpha is not None: 
            self.alpha = None 
            warnings.warn("include_background=False, alpha ignored when using softmax.") 
        loss = softmax_focal_loss(input, target, self.gamma, self.alpha)
``` 

This is the code piece. Not sure why when we don't include background and when using softmax, alpha became invalid? Does this assume there are no label imbalance? How to address this when there is significantly more negative samples than positive samples in the experiment then?

**Comment by KumoLiu** [comment]
Hi @dancingreindeer, we only calculate it for background class and other classes, so If include_background is set to False, there will be no background class to calculate, then set it to None. You can set `weight` to apply to each class as well.
https://github.com/Project-MONAI/MONAI/blob/acfc508ebd35ad46bbccbd19f2eb57d34609f0e7/monai/losses/focal_loss.py#L218

Thanks.

**Reply by dancingreindeer** [reply]
@KumoLiu Thank you for your answer. However, when include_background is set to True, shouldn't we be given an option to assign different alpha value to different class? For example, for voxels with label index 0, we can assign an alpha = 0.1, for other classes, we can assign 0.7. 

I understand this alpha is different than the class weights itself. What the alpha tried to capture is how much emphasize being put on the voxel depending on whether or not it is majority or minority in the samples. I am just thinking different classes may have different importance for their foreground voxels?

i can see the "weight" could play similar roles though.
